rfd/0097-connect-usage-metrics.md

```diff
@@ -0,0 +1,221 @@
+state: draft
+ ---
+
+# RFD 97 - Collecting Teleport Connect usage metrics
+
+## Required Approvers
+
+* Engineering:? @ravicious @zmb3
+* Product: @klizhentas @xinding33
+
+## What
+
+Collect downloads and usage metrics of Teleport Connect.
+
+## Why
+
+Currently, the team has no information on how many users download, install and use Teleport Connect on a daily basis.
+In order to effectively plan the development of the product, the team should also know what is the adoption of new
+features, which ones are the most popular and which are problematic for users.
+
+## Details
+
+Connect will have a stateless metrics service that will send events to an unauthorized gRPC endpoint, exposed by TSH
+daemon. The anonymization should be done in `tshd` (to be open on how it is done and what is sent further). Events will
+be sent to `prehog` directly.
+It was considered to use authorized endpoint provided by cluster's `Auth Service`, but it seems to not work well for
+Connect for a few reasons:
+
+- Some events do not belong to any cluster, like _app start_ event.
+- A user can use Connect while not logged in. For example, new tabs can be opened after the session expires.
+
+Since Connect will use endpoint that is not aware of the cluster context, it must provide some value that will
+differentiate events coming from multiple application instances. To do this, each event will have `distinct_id` field.
+It will be supplied with UUID generated by Connect, which will be generated once and stored in a file in the _App Data_
+directory, so it won't change between restarts.
+
+### User agreement
+
+On the start, Teleport Connect will ask user to opt in to volunteer anonymized metrics and usage-data with standard
+message "Are you OK sending anonymized usage data about Teleport Connect? This will help us to improve product".
+
+If the user refuses, no data will be collected.
+
+### How will collecting metrics support product development?
+
+In the initial version, it should help with getting answers to the following questions:
+
+#### How many unique users download and use Teleport Connect today?
+
+To answer the first part of the question, download counts from `goteleport.com/download` are needed.
+This task should be done by the marketing team.
+
+To calculate how many users use Teleport Connect on a daily basis, a metric like DAU (Daily Active Users) can be used.
+This metric can be based on a specific event, but in this case it should be calculated using any event. For example,
+user logged only once in a given day to refresh certs for a DB proxy connection - such user can be considered as active.
+
+#### What features are the most popular?
+
+Usage of each feature will be measured basing on events from the [events](#events) section. They will allow to generate
+various statistics, like the most common opened kinds of tabs or just show the usage of particular UI elements.
+
+#### Where users struggle the most?
+
+//TODO
```

Comment:
I don't have a good idea how to measure it.

For example, we can measure the time from opening the “Request Access” tab to sending the actual access request. 
But is it really useful and can be applied to other places?